processmanager/processinfo.go

```diff
@@ -730,3 +746,14 @@ func (pm *ProcessManager) ProcessedUntil(traceCaptureKTime times.KTime) {
 
 // Compile time check to make sure we satisfy the interface.
 var _ tracehandler.TraceProcessor = (*ProcessManager)(nil)
+
+func (pm *ProcessManager) EnvVarsForPID(pid libpf.PID) map[string]string {
+	var envVars map[string]string
+
+	pm.mu.RLock()
```

Comment:
How critical are information about environment variables, that they should hold a global lock on the ProcessManager?

Reply:
When a user requires this information, it has the same criticality then data from other *ForPID functions.

In `loadBpfTrace()` we call now three of these functions when initializing the `host.Trace{}` instance. This means we do an RLock three times.

I'd suggest to fetch the data with a single function call / single lock in a follow-up PR.